src/openms/source/FORMAT/HANDLERS/MzMLSqliteHandler.cpp

```diff
@@ -0,0 +1,981 @@
+        if (sqlite3_column_type(stmt, 9) != SQLITE_NULL) 
+        {
+          const unsigned char * pepseq = sqlite3_column_text(stmt, 9);
+          peptide_sequence = std::string(reinterpret_cast<const char*>(pepseq), sqlite3_column_bytes(stmt, 9));
+          precursor.setMetaValue("peptide_sequence", peptide_sequence);
+        }
+        // if (sqlite3_column_type(stmt, 10) != SQLITE_NULL) product.setCharge(sqlite3_column_int(stmt, 10));
+        if (sqlite3_column_type(stmt, 11) != SQLITE_NULL) product.setMZ(sqlite3_column_double(stmt, 11));
+        if (sqlite3_column_type(stmt, 12) != SQLITE_NULL) product.setIsolationWindowLowerOffset(sqlite3_column_double(stmt, 12));
+        if (sqlite3_column_type(stmt, 13) != SQLITE_NULL) product.setIsolationWindowUpperOffset(sqlite3_column_double(stmt, 13));
+        if (sqlite3_column_type(stmt, 14) != SQLITE_NULL) 
+        {
+          int pol = sqlite3_column_int(stmt, 14);
+          if (pol == 0) spec.getInstrumentSettings().setPolarity(IonSource::NEGATIVE);
+          else spec.getInstrumentSettings().setPolarity(IonSource::POSITIVE);
+        }
+        if (sqlite3_column_type(stmt, 15) != SQLITE_NULL && sqlite3_column_int(stmt, 15) != -1
+            && sqlite3_column_int(stmt, 15) < static_cast<int>(OpenMS::Precursor::SIZE_OF_ACTIVATIONMETHOD))
+        {
+          precursor.getActivationMethods().insert(static_cast<OpenMS::Precursor::ActivationMethod>(sqlite3_column_int(stmt, 15)));
+        }
+        if (sqlite3_column_type(stmt, 16) != SQLITE_NULL) precursor.setActivationEnergy(sqlite3_column_double(stmt, 16));
+
+        if (sqlite3_column_type(stmt, 6) != SQLITE_NULL) spec.getPrecursors().push_back(precursor);
+        if (sqlite3_column_type(stmt, 11) != SQLITE_NULL) spec.getProducts().push_back(product);
+        spectra.push_back(spec);
+
+        sqlite3_step( stmt );
+      }
+
+      // free memory
+      sqlite3_finalize(stmt);
+    }
+
+    void MzMLSqliteHandler::writeExperiment(const MSExperiment<> & exp)
+    {
+      writeChromatograms(exp.getChromatograms());
+      writeSpectra(exp.getSpectra());
+    }
+
+    void MzMLSqliteHandler::createTables()
+    {
+      sqlite3 *db;
+      char *zErrMsg = 0;
+      int rc;
+
+      // delete file if present
+      QFile file (filename_.toQString());
+      file.remove();
+
+      // Open database
+      rc = sqlite3_open(filename_.c_str(), &db);
+      if (rc)
+      {
+        throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, String("Can't open database: ") + sqlite3_errmsg(db));
+      }
+
+      // Create SQL structure
+      char const *create_sql =
+
+        // data table
+        //  - compression is one of 0 = no, 1 = zlib, 2 = np-linear, 3 = np-slof, 4 = np-pic, 5 = np-linear + zlib, 6 = np-slof + zlib, 7 = np-pic + zlib
+        //  - data_type is one of 0 = mz, 1 = int, 2 = rt
+        //  - data contains the raw (blob) data for a single data array
+        "CREATE TABLE DATA(" \
```

Comment:
see below, I dont really think that it would from a DB design perspective. Also it maps well to mzML ...